models/DataObject/ClassDefinition/Data/ManyToOneRelation.php

```diff
@@ -631,6 +631,10 @@ public function getReturnTypeDeclaration(): ?string
      */
     public function addListingFilter(DataObject\Listing $listing, $data, $operator = '=')
     {
+        if ($data === null) {
+            return $listing;
```

Comment:
The question is whether this is the desired behaviour. When I would execute a

```
$cars = (new \Pimcore\Model\DataObject\Car\Listing())->filterByManufacturer(null);
```

then I would actually expect a listing of all cars where the Manufacturer relation is empty. The current suggested implementation returns just all cars. It does solve your bug, but IMHO doesn't make the product better.

Also note (PHPStan is complaining about this) that both the

`\Pimcore\Model\DataObject\ClassDefinition\Data::addListingFilter()`
and all the auto generated
`\Pimcore\Model\DataObject\*\Listing::filterBy*()`
methods have an argument signature

```
* @param string|int|float|array|Model\Element\ElementInterface $data comparison data, can be scalar or array (if operator is e.g. "IN (?)")
```
so `NULL` isn't an argument type that is supported by that method.

Either that needs to be updated (that those methods really support null, but then we would need a proper null handling as well) or we should explicitly disallow null values (InvalidArgumentException or so).
Pimcore 11.x is currently implemented as the first choice (so that needs to be taken into account as well, that if we change it here, then 11 should be updated as well): https://github.com/pimcore/pimcore/blob/11.x/models/DataObject/ClassDefinition/Data.php#L1301

Reply:
@youwe-petervanderwal thank you for reviewing.
Now, I noticed that non-relational fields also have the same behavior. As you said, it is not a bug.
I am closing this PR. It can break listings. Doesn't worth it.